File: backend/src/main/java/com/new_cafe/app/backend/entity/Menu.java
package com.new_cafe.app.backend.entity;

import java.time.LocalDateTime;

import lombok.Data;
import lombok.Builder;
import lombok.AllArgsConstructor;
import lombok.NoArgsConstructor;



@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Menu {
    private Long id;
    private String korName;
    private String engName;
    private String description;
    private Integer price;
    // private String categoryName;
    private Long categoryId;
    // private String imageSrc;
    private Boolean isAvailable;
    // private Boolean isSoldOut;
    // private Integer sortOrder;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;

    private Category category;
    // private List<MenuImage> images;

    
}

File: frontend/types/menu.ts
// types/menu.ts

export interface Menu {
  id: number;
  korName: string;
  engName: string;
  description: string;
  price: number;
  category: MenuCategory;
  categoryId?: number;
  images: MenuImage[];
  isAvailable: boolean;
  isSoldOut: boolean;
  sortOrder: number;
  options: MenuOption[];
  createdAt: Date;
  updatedAt: Date;
}

export interface MenuImage {
  id: string;
  url: string;
  isPrimary: boolean;
  sortOrder: number;
}

export interface MenuOption {
  id: string;
  name: string;
  type: 'radio' | 'checkbox';
  required: boolean;
  items: OptionItem[];
}

export interface OptionItem {
  id: string;
  name: string;
  priceDelta: number;
}

export interface MenuCategory {
  id: number;
  korName: string;
  engName: string;
  sortOrder: number;
}

// Form types
export interface MenuFormData {
  korName: string;
  engName: string;
  description: string;
  price: number;
  categoryId: number;
  isAvailable: boolean;
  images: File[];
  options: MenuOption[];
}

// Filter types
export type MenuStatusFilter = 'all' | 'available' | 'soldout' | 'hidden';
